main.js

```diff
@@ -0,0 +1,159 @@
+console.log(`Загальна кількість товарів = ${totalQantity}.`)
+
+// Знайти найдорожчий продукт
+const mostExpensiveProduct = productArray.sort(sortedByField("productPrice"));
+console.log("Найдорожчий продукт, який ми маємо придбати - " + mostExpensiveProduct[mostExpensiveProduct.length-1].productName + ", продукт коштує - " + mostExpensiveProduct[mostExpensiveProduct.length-1].productPrice + " грн.")
+
+// Створити функцію, яка повинна додавати новий продукт в існуючий масив
+const newProduct = {};
+
+const addNewProduct = (array, obj) => {
+    obj.productName = prompt("Введіть назву продукту, який би додатково хотіли б придбати.");
+    obj.productPrice = +prompt("Вкажіть вартість цього продукту.");
+    obj.productQuantity = +prompt("Вкажіть кількість цього продукту.");
+    if (VALIDATE(obj.productName) || !isNaN(obj.productName) || VALIDATE(obj.productPrice) || validateNumbers(obj.productPrice) || VALIDATE(obj.productQuantity) || validateNumbers(obj.productQuantity)){
+        return false;
+    } else {
+        return array.push(obj);
+    }
+};
+addNewProduct(productArray, newProduct);
+console.log(productArray);
+
+
+// Створити функцію, яка повинна видаляти конкретний продукт із існуючий масиву продуктів
+// Дані про видалення такого продукту - користувач вводить із prompt лише назву
+const deleteProduct = (array) => {
+    const userProduct = prompt("Введіть назву продукту, який би ви хотіли прибрати з корзини");
+    if(VALIDATE(userProduct)) {
+        return false;
+    } else {
+        for (let i = 0; i < array.length; i++) {
+            if(typeof(array[i]) === "object" && array[i].productName === userProduct) {
+                array.splice(i, 1);
+            }
+        }
+        return array;
+    } 
+};
+deleteProduct(productArray);
+console.log(productArray);
+
+
+//4. Дано масив [16, -3, 54,-4, 72,-56, 47, -12, 4, -16, 25, -37, 46, 4, -51, 27, -8, 4, -54, 76, -4, 12, 6, -35]
+
+let myArray = [16, -3, 54,-4, 72,-56, 47, -12, 4, -16, 25, -37, 46, 4, -51, 27, -8, 4, -54, 76, -4, 12, 6, -35];
+
+// Знайти суму та кількість позитивних елементів.
+let positiveNumbers = 0;
+for (let i = 0; i < myArray.length; i++) {
+    if (myArray[i] > 0) {
+        positiveNumbers += myArray[i];
+    }
+}
+console.log(`Сума позитивних елементів = ${positiveNumbers}.`);
+
+function getPositiveCount(arr) {
+    return arr.filter(el => el > 0).length;
+};
+console.log("Кількість позитивних елементів: " + getPositiveCount(myArray) + ".");
+
+// Знайти мінімальний елемент масиву та його порядковий номер.
+const min = Math.min(...myArray);
+const positionMinInArray = myArray.indexOf(min) + 1;
+console.log(`Мінімальний елемент масиву: ${min}, його порядковий номер: ${positionMinInArray}.`);
+
+// Знайти максимальний елемент масиву та його порядковий номер.
+const max = Math.max(...myArray);
+const positionMaxInArray = myArray.indexOf(max) + 1;
+console.log(`Mаксимальний елемент масиву: ${max}, його порядковий номер: ${positionMaxInArray}.`);
```

Comment:
```suggestion
console.log(`Максимальний елемент масиву: ${max}, його порядковий номер: ${positionMaxInArray}.`);
```

Reply:
Can you please explain what I have to change in my code? In "suggested changes" I see two similar strings.

Reply:
Here is also a typo, maybe based on languages
<img width="782" alt="image" src="https://user-images.githubusercontent.com/84010501/193536530-a7683875-4bc2-420c-b2d1-e23b3ea14b99.png">